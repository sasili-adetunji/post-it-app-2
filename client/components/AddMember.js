import React from 'react';
import { CardHeader, CardTitle } from 'material-ui/Card';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Paper from 'material-ui/Paper';
import PostItActions from '../actions/PostItActions';
import PostItStore from '../stores/PostItStore';
import _ from 'lodash';

let userName;

const style = {
  height: 100,
  width: 100,
  margin: '30px auto',
  padding: 30,
  textAlign: 'center',
  display: 'inline-block',
};
/**
 * Addmember component.
 * @returns {String} The markup for the addMember component
 */

class AddMember extends React.Component {
  /**
   * Creates an instance of AddMember.
   * @param {any} props
   * @memberof AddMembe
   */
  constructor(props) {
    super(props);
    this.state = {
      groupname: '',
      userId: '',
      users: PostItStore.getUsers()
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
    this.userName = this.userName.bind(this);
    // this.makeUser = this.makeUser.bind(this);
    // this.users = this.users.bind(this);
  }

  /**
   * Monitors changes in the components and change the state
   * @param {any} e
   * @memberof AddMember
   */
  onChange(e) {
    e.preventDefault();
    // this.setState({
    //   [e.target.name]: e.target.value
    // });
    const user = _.map(this.state.users).map((x) => {
      console.log(x.username, '???????????');
      return x.username;
    });
  }

  /**
   * makes an action to add member
   *
   * @param {any} e
   * @memberof AddMember
   */
  onClick(e) {
    e.preventDefault();
    const user = {
      userId: this.state.userId,
      groupId: this.state.groupId
    };
    PostItActions.addUserToGroup(user);
  }
  // userName() {
  //   console.log('xxxxxxx');
  //   const user = _.map(this.state.users).map((x) => {
  //     console.log(x.username, '???????????');
  //     return x.username;
  //   });
  // }
  userName(e) {
    e.preventDefault();
    let userss = '';
    const user = _.map(this.state.users).map((x) => {
      userss = x.username;
    });
    console.log(userss, 'hhhhhhhh');
    // userss.map((name, index) => {
    //   return <option key={index}>{name} </option>;
    // });
  }
  makeUser() {
    console.log('xxxxxxx');
    (X) => {
      console.log(X, 'xxxxxxx');
      return this.userName.map(<option> {X} </option>);
    };
  }

  // users() {
  //   console.log(this.userName.map(this.makeUser), 'usssssssss');
  //   return this.userName.map(this.makeUser);
  // }


  /**
   * renders the addMember components
   *
   * @returns {void}
   * @memberof AddMember
   */
  render() {
    const user = _.map(this.state.users).map((x) => {
      return x.username;
    });
    return (
      // <TextField
      //      name="userId" onChange={this.userName}
      //   value={this.state.userId}
      //      floatingLabelText="User ID" />
      <select onChange={this.onChange} >
        {user.map((name, index) => {
          return <option value="user" key={index}>{name} </option>;
        })}
      </select>
    );
    // return (

    //   <div style={style}>
    //     <CardTitle title="Add" />
    //     <TextField
    //       name="groupId" onChange={this.onChange} value={this.state.groupId}
    //       floatingLabelText="Group ID" /><br />
    //     <TextField
    //       name="userId" onChange={this.onChange} value={this.state.userId}
    //       floatingLabelText="User ID" /><br />
    //     <select onChange={this.onChange}>
    //       <option value="user">{this.userName}</option>
    //     </select>
    //     <RaisedButton
    //       style={{
    //         display: 'block',
    //         width: '20px'
    //       }}
    //       onClick={this.onClick}
    //       label="Add" primary />
    //   </div>
    // );
  }

}

export default AddMember;
