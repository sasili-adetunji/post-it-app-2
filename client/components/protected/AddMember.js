import React from 'react';
import lodash from 'lodash';
import PostItStore from '../../stores/PostItStore';
import PostItActions from '../../actions/PostItActions';


/**
 * creates addmember components
 *
 * @class AddMember
 * @extends {React.Component}
 */
class AddMember extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      userId: '',
      error: ''
    };
    // this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
    this.data = this.data.bind(this);
  }

  // onChange(e) {
  //   this.setState({
  //     [e.target.name]: e.target.value
  //   });
  // }

  /**
   * function that get userid from username
   *
   * @param {any} userName
   * @returns
   * @memberof AddMember
   */
  data(userName) {
    let n;
    lodash.map(this.props.usern).map((x) => {
      if (userName === x.userName) {
        n = x.userId;
      } else {
        return null;
      }
    });
    return n;
  }
  onClick(event) {
    event.preventDefault();
    if (!this.props.selected[0]) {
      this.setState({
        error: 'Kindly select a group first',
      });
      this.refs.add.value= '';
      return true;
    }
    const user = {
      userId: this.data(this.refs.add.value.trim()),
      userName: this.refs.add.value.trim(),
      groupId: this.props.selected[0].groupId
    };

    if (!user.userId) {
      this.setState({
        error: 'This User does not exist'
      });
      this.refs.add.value= '';
    } else {
      PostItActions.addUserToGroup(user);
      // console.log(user);
      this.refs.add.value = '';
      this.setState({
        error: ''
      });
    }
  }
  /**
   *
   * renders add member components
   * @returns { void }
   * @memberof AddMember
   */
  render() {
    return (
      <div className="panel-body">
        <h6> To add a member, type in the username of the member </h6>
            <strong className="error"> {this.state.error} </strong> 
        <form className="navbar-form" role="search">
          <div className="form-group">
            <input
              type="text" ref="add" className="form-control" placeholder="Add member"
              />
          </div>
          <button onClick={this.onClick} type="submit" className="btn btn-default ">
            <span className="glyphicon glyphicon-plus" /></button>
        </form>
        <br />
      </div>
    );
  }
}
export default AddMember;
