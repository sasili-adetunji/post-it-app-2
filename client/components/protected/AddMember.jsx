import React from 'react';
import lodash from 'lodash';
import PostItStore from '../../stores/PostItStore';
import PostItActions from '../../actions/PostItActions';


/**
 * creates addmember components
 *
 * @class AddMember
 * 
 * @extends {React.Component}
 */
class AddMember extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      userId: '',
      error: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
    this.changeToUserId = this.changeToUserId.bind(this);
  }


  /**
    * @method onChange
    *
    * @description Monitors changes in the components and change the state
    *
    * @memberof AddMember
    *
    * @param {object}
    *
    * @returns {void}
    */
  onChange(event) {
    this.setState({
      userName: event.target.value,
    });
  }


  /**
   * @description function that get userid from username
   *
   * @param {String} userName 
   * 
   * @memberof AddMember
   */
  changeToUserId(userName) {
    let userId;
    lodash.map(PostItStore.getUsers()).map((user) => {
      if (userName === user.userName) {
        userId = user.userId;
      } else {
        return null;
      }
    });
    return userId;
  }

   /**
     * @description Makes an action call to add a member to a group
     * 
     * @param {object} event
     * 
     * @returns {void}
     * 
     * @memberof AddMember
  */

  onClick(event) {
    event.preventDefault();
    if (!PostItStore.getOpenedGroup()[0]) {
      this.setState({
        error: 'Kindly select a group first',
        userName: '',
      });
      return true;
    }
    const user = {
      userId: this.changeToUserId(this.state.userName),
      userName: this.state.userName,
      groupId: PostItStore.getOpenedGroup()[0].groupId,
    };
    if (!user.userId) {
      this.setState({
        error: 'This User does not exist',
        userName: '',
      });
    } else {
      PostItActions.addUserToGroup(user);
      this.setState({
        error: '',
        userName: '',
      });
    }
  }
 /**
   * @method render
   * 
   * Render addmember component
   *
   * @returns {String} The HTML markup for the AddMember Components
   * 
   * @memberof AddMember
   */
  render() {
    return (
      <div className="panel-body">
        <h6> To add a member, type in the username of the member </h6>
            <strong className="error"> {this.state.error} </strong>
        <form className="navbar-form" role="search">
          <div className="form-group">
            <input
              type="text" className="form-control" placeholder="Add member" 
              name="userName"
              onChange={this.onChange} value={this.state.userName}
              />
          </div>
          <button onClick={this.onClick} type="submit" 
          className="btn btn-default ">
            <span className="glyphicon glyphicon-plus" /></button>
        </form>
        <br />
      </div>
    );
  }
}
export default AddMember;
