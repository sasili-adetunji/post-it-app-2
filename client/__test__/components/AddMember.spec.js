import React from 'react';
import { shallow, render } from 'enzyme';
import AddMember from '../../components/protected/AddMember.jsx';


function setup() {
  const props = {
    onClick: () => {},
    onChange: () => {},

  };
  return shallow(<AddMember />);
}

describe('AddMember components', () => {
  it('should match snapshot test', () => {
    const component = shallow(<AddMember />);
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    const component = shallow(<AddMember />);
    expect(component).toBeDefined();
  });
  it('Should contain two div', () => {
    const component = shallow(<AddMember />);
    expect(component.find('div').length).toEqual(2);
  });
  it('should recieve props', () => {
    const component = shallow(<AddMember />);
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
});

describe('AddMember  Test', () => {
  it('should take props', () => {
    const wrapper = setup();
    expect(wrapper.props().onClick).toExist;
    expect(wrapper.props().onChange).toExist;

  });
});
