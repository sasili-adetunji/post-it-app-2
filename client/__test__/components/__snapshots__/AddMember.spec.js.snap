// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMember components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="panel-body"
  >
    <h6>
       To add a member, type in the username of the member 
    </h6>
    <strong
      className="error"
    >
       
      
       
    </strong>
    <form
      className="navbar-form"
      role="search"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          name="userName"
          onChange={[Function]}
          placeholder="Add member"
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-default "
        onClick={[Function]}
        type="submit"
      >
        <span
          className="glyphicon glyphicon-plus"
        />
      </button>
    </form>
    <br />
  </div>,
  "nodes": Array [
    <div
      className="panel-body"
    >
      <h6>
         To add a member, type in the username of the member 
      </h6>
      <strong
        className="error"
      >
         
        
         
      </strong>
      <form
        className="navbar-form"
        role="search"
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="userName"
            onChange={[Function]}
            placeholder="Add member"
            type="text"
            value=""
          />
        </div>
        <button
          className="btn btn-default "
          onClick={[Function]}
          type="submit"
        >
          <span
            className="glyphicon glyphicon-plus"
          />
        </button>
      </form>
      <br />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <AddMember />,
    "_instance": AddMember {
      "context": Object {},
      "data": [Function],
      "onChange": [Function],
      "onClick": [Function],
      "props": Object {},
      "refs": Object {},
      "state": Object {
        "error": "",
        "userId": "",
        "userName": "",
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="panel-body"
    >
      <h6>
         To add a member, type in the username of the member 
      </h6>
      <strong
        className="error"
      >
         
        
         
      </strong>
      <form
        className="navbar-form"
        role="search"
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="userName"
            onChange={[Function]}
            placeholder="Add member"
            type="text"
            value=""
          />
        </div>
        <button
          className="btn btn-default "
          onClick={[Function]}
          type="submit"
        >
          <span
            className="glyphicon glyphicon-plus"
          />
        </button>
      </form>
      <br />
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AddMember />,
}
`;
